File: java_brain/src/main/java/org/netbrains/javabrain/neural/Connection.java
package org.netbrains.javabrain.neural;

public class Connection {
    private double weight;
    private final Neuron inputNeuron;

    public Connection(Neuron inputNeuron) {
        this.inputNeuron = inputNeuron;
        weight = (Math.random() -0.5) * 2.0;
    }

    public double getWeight() {
        return weight;
    }

    public Neuron getInputNeuron() {
        return inputNeuron;
    }

    public double activate() {
        return inputNeuron.activate() * weight;
    }

    public void adjustWeight(double delta) {
        weight += delta;
    }
}

File: swift_brains/swift_brains/Connection.swift

class Connection {
    private let inputNeuron: Neuron
    
    private var weight: Double = Double.random(in: 0...1)
    
    init (inputNeuron: Neuron) {
        self.inputNeuron = inputNeuron
    }
    
    public func getInputNeuron() -> Neuron {
        return inputNeuron
    }
    
    public func getWeight() -> Double {
        return weight
    }
    
    public func adjustWeight(delta: Double) {
        weight += delta
    }
}
